src/sgame/sg_buildable.cpp

```diff
@@ -1702,110 +1702,97 @@ itemBuildError_t G_CanBuild( gentity_t *ent, buildable_t buildable, int /*distan
 }
 
 /** Sets shared buildable entity parameters. */
-#define BUILDABLE_ENTITY_SET_PARAMS(params)\
-	params.oldEnt = ent;\
+template <typename entityType>
+void BUILDABLE_ENTITY_CREATE( gentity_t *ent, buildable_t buildable )
+{
+	typename entityType::Params params;
+	params.oldEnt = ent;
 	params.Health_maxHealth = BG_Buildable(buildable)->health;
+	ent->entity = new entityType(params);
+}
 
-#define BUILDABLE_ENTITY_START(entityType)\
-	entityType::Params params;\
-	BUILDABLE_ENTITY_SET_PARAMS(params);
-
-#define BUILDABLE_ENTITY_END(entityType)\
+template < typename entityType, typename std::enable_if<
+	std::is_same<entityType,DrillEntity>::value ||
```

Comment:
This will never compile as written (`void = 0`)

Reply:
Wdym? This actually passes my local build and the CI, so I guess you're not talking about compiler itself?

Reply:
> This actually passes my local build and the CI, so I guess you're not talking about compiler itself?

Yes I expect the build to pass, but *this template specialization* will never compile. With the `= 0` or `= true` type examples you need to specify a 2nd argument for `enable_if` so that `type` is something that can be 0.